### ISSUE TITLE
Loading global files simultaneously with production build

### ISSUE THREAD

**Issue by andyrooger** [systemjs/systemjs#1810]
We are using SystemJS to load our test specs. This includes a mixture of older global files and newer es6 and system modules. We're loading these with the following code.

```
var specFiles = [...];
Promise.all(specFiles.map(s => System.import(s)))
    .then(() => {
        // run tests
    });
```

Using the standard SystemJS build this works fine. Unfortunately, with the production build it does not. ES modules still load, however only the first global file loads correctly. All subsequent global files fail.

We get a different error depending on the browser - in Chrome it looks like

> Error: Cannot read property 'postMessage' of undefined

From debugging, it looks like a problem with working out the exports for the global files.
* There seems to be one instance of `globalSnapshot` which is initialised over and over again when the imports are kicked off. 
* The first import completes correctly and resets `globalSnapshot`
* All the other imports fail since their initial snapshot has been reset to `undefined`

I've created a small repro here: https://plnkr.co/edit/KWmtqd4CVIkFtfl6gyNJ?p=preview

**Comment by guybedford** [comment]
Thanks so much for the clear replication here, fixed in https://github.com/systemjs/systemjs/commit/5ce46c325430d07ecac925441b6f04061ca258a9.

**Comment by andyrooger** [comment]
That was fast! Thanks @guybedford

**Comment by guybedford** [comment]
Released in 0.21.2 :)
